registrations/serializers.py

```diff
@@ -433,6 +449,82 @@ def validate(self, data):
         data = super().validate(data)
         return data
 
+    @staticmethod
+    def send_notifications(signup_instances):
+        confirmation_type_mapping = {
+            SignUp.AttendeeStatus.ATTENDING: SignUpNotificationType.CONFIRMATION,
+            SignUp.AttendeeStatus.WAITING_LIST: SignUpNotificationType.CONFIRMATION_TO_WAITING_LIST,
+        }
+
+        for signup in signup_instances:
+            if not signup.responsible_for_group and signup.signup_group_id:
+                continue
+
+            signup.send_notification(confirmation_type_mapping[signup.attendee_status])
+
+    def create_signups(self, validated_data):
+        user = self.context["request"].user
+        registration = validated_data["registration"]
+
+        add_as_attending, add_as_waitlisted = _validate_registration_capacity(
+            registration, len(validated_data["signups"])
+        )
+        capacity_by_attendee_status_map = {
+            SignUp.AttendeeStatus.ATTENDING: add_as_attending,
+            SignUp.AttendeeStatus.WAITING_LIST: add_as_waitlisted,
+        }
+
+        def check_and_set_attendee_status(cleaned_signup_data):
+            if capacity_by_attendee_status_map.get(
+                cleaned_signup_data.get("attendee_status")
+            ):
+                return
+
+            if capacity_by_attendee_status_map[SignUp.AttendeeStatus.ATTENDING]:
+                cleaned_signup_data["attendee_status"] = SignUp.AttendeeStatus.ATTENDING
+            else:
+                cleaned_signup_data[
+                    "attendee_status"
+                ] = SignUp.AttendeeStatus.WAITING_LIST
+
+        signups = []
+        protected_data = []
+
+        for signup_data in validated_data["signups"]:
+            if not (
+                capacity_by_attendee_status_map[SignUp.AttendeeStatus.ATTENDING]
+                or capacity_by_attendee_status_map[SignUp.AttendeeStatus.WAITING_LIST]
+            ):
+                continue
```

Comment:
break?

Reply:
Yes indeed, that is better and has been changed as suggested now.